staging/src/k8s.io/component-base/config/v1alpha1/defaults.go

```diff
@@ -44,9 +45,7 @@ func RecommendedDefaultLeaderElectionConfiguration(obj *LeaderElectionConfigurat
 		obj.RetryPeriod = metav1.Duration{Duration: 2 * time.Second}
 	}
 	if obj.ResourceLock == "" {
-		// TODO(#80289): Figure out how to migrate to LeaseLock at this point.
```

Comment:
I don't think we can change defaulting logic for existing API version - can we?

@liggitt - for your comments